### ISSUE TITLE
Rule Proposal: Disallow certain extensions in import path

### ISSUE THREAD

**Issue by lo1tuma** [benmosher/eslint-plugin-import#209]
When I transpile my `.jsx` files with babel they will be renamed to have a `.js` extension. That’s why I usually avoid the `.jsx` extentions in the import path, but sometimes other contributors of my project commit code that has import statements with file extensions.

Having an eslint rule would be perfect to identify such problems.

When the rule is configured with: `no-import-file-extensions: [ 2, [ '.js', '.jsx', '.json' ] ]` this would be considered as problems:

```js
import Foo from './Foo.jsx';
import lib from './lib.js';
import data from './data.json';
```

This would not be considered as a problem:

```js
import Foo from './Foo';
import lib from './lib';
import data from './data';
```

**Comment by benmosher** [comment]
Interesting. This is almost the converse of #204. 

Makes me think maybe a rule could support both by either enforcing extensions (`import/extensions: [2, 'always']` for #204) or rejecting them, for this (`import/extensions: [2, never: [js, jsx, json]]`).

For my memory: may want to stat paths that fail to check that they don't exist without an extension (i.e. directories)
